frontend/src/state/appGlobal.ts

```diff
@@ -10,26 +10,55 @@
  */
 
 import type { History } from 'history';
+import type { Location, NavigateFunction } from 'react-router-dom';
 import { api } from './backendApi';
 import { uiState } from './uiState';
 
 class AppGlobal {
-  private _history = null as unknown as History<any>;
-  get history() {
-    return this._history;
+  private _navigate = null as unknown as NavigateFunction;
+  private _location = null as unknown as Location;
```

Comment:
Do we need to map every single part of history/routing? For example ability to go back

Reply:
I replaced the `.goBack()` by `.navigate(-1)` - I think there were only one occurrence of that.

Eventually I'd like to get rid of the the `appGlobal` entirely, but we will have to migrate all the class-based components to functional components first.

Reply:
No worries. Incremental adoption is ok